light-cv:0.1.0

AnsgarLichter opened the pull request:
I am submitting
- [x] a new package
- [ ] an update for a package

Description: It's a rather minimalistic CV which I use for my own application process. I just wanted to share this template with the typst world to help others.

I have read and followed the submission guidelines and, in particular, I
- [x] selected a name that isn't the most obvious or canonical name for what the package does
- [x] added a `typst.toml` file with all required keys
- [x] added a `README.md` with documentation for my package
- [x] have chosen a license and added a `LICENSE` file or linked one in my `README.md`
- [x] tested my package locally on my system and it worked
- [x] `exclude`d PDFs or README images, if any, but not the LICENSE

- [x] ensured that my package is licensed such that users can use and distribute the contents of its template directory without restriction, after modifying them through normal use.

reknih commented on packages/preview/light-cv/0.1.0/README.md:
```diff
@@ -0,0 +1,29 @@
+
+This is my CV template written in Typst. You can find a two example CVs in this repository as PDFs:
+
+- [German CV](cv-de.pdf)
+- [English CV](cv-de.typ)
```
These links will not work on Universe. Please change them to absolute links or replace them.

reknih commented on packages/preview/light-cv/0.1.0/cv-en.typ:
```diff
@@ -0,0 +1,167 @@
+)
+
+#header(
+  fullName: [John Doe],
+  jobTitle: [Software Engineer with a passion for JavaScript],
```
For Typst, we normally use `kebab-case`.

reknih commented on packages/preview/light-cv/0.1.0/fontAwesome/Font Awesome 6 Brands-Regular-400.otf:
Since you are depending on the Font Awesome package, this can be removed.

reknih commented on packages/preview/light-cv/0.1.0/typst.toml:
```diff
@@ -0,0 +1,17 @@
+repository = "https://github.com/AnsgarLichter/cv-typst-template"
+keywords = ["cv", "curriculum vitae", "computer science", "application", "resume"]
+categories = ["cv"]
+disciplines = ["engineering", "computer-science"]
```
Can this CV be used by professionals in other disciplines? If so, remove the disciplines key.

reknih commented on packages/preview/light-cv/0.1.0/README.md:
It could be a good idea to describe what the parameters of `header` as well as the `section` and `entry` functions do in the README

reknih commented on packages/preview/light-cv/0.1.0/template.typ:
```diff
@@ -0,0 +1,157 @@
+
+#import "settings/styles.typ": *
+
+#import "modules/utils.typ": *
+#import "modules/header.typ": *
+#import "modules/section.typ": *
+#import "modules/skills.typ": *
```
Normally, the files you stored within the `modules` dictionary are not copied into the user project. Instead, they are imported through the package manager.

I would suggest that you move your settings folder and `cv-en.typ` to a new folder `template` and make that directory the template's path in the `typst.toml`. Then, you can create a new `lib.typ` and reexport all relevant functions from the `template.typ` and the files in `modules` there and make it your entrypoint.

For orientation, I have attached a screenshot of the directory structure your template generates when used with `typst init` and how it looks like for other templates below: 

![Finder screenshot of the directory structure of `starter-journal-article` and `light-cv`](https://github.com/typst/packages/assets/3874949/65eb6095-dcbd-4fbf-963f-53c25b12a3b3)

reknih reviewed (commented):
Thank you for your submission. I have added a few comments that should help to get this accepted!

AnsgarLichter commented:
Thanks, I appreciate your feedback. I will do the necessary changes in the near future, may take some time due to easter holidays and so on.  Please just don't close this PR.

laurmaedje commented:
No worries, we won't close it. Thanks for the update. Enjoy your holidays!

AnsgarLichter commented:
I finally included all your feedback with my recent commits:
- updated the links to the samples to absolute links
- use `kebab-case` consistently
- remove the fonts from front-awesome
- remove disciplines
- update the structure and adapt the `typst.toml` accordingly

I registered the package as a local package on my machine. `typst init @preview/light-cv:0.1.0` results in the following directory structure:
- `light-cv`
    - `settings`
        - `styles.typ`
    - `cv-en.typ`

So if I execute `cd light-cv` and `typst watch cv-en.typ` the pdf is created successfully.
If you have any further feedback please let me know 👍

laurmaedje reviewed (commented):
The previous review comment about the fontawesome package wasn't 100% accurate. As embedded, the font didn't do anything, but the `fontawesome` package also doesn't ship with the font. I would add a note to the README about where the user can obtain the font and then they'll need to download + install/upload it themselves.

AnsgarLichter commented:
Thanks for your detailed feedback and explanation again.
I applied the following changes:
- removed example image from README and package,
- explained font awesome installation by linking to fontawesome pacakge's README on Typst Universe,
- Refactored image handling

AnsgarLichter commented on packages/preview/light-cv/0.1.0/README.md:
```diff
@@ -0,0 +1,48 @@
+![alt text](media/image.png)
```
Done

AnsgarLichter commented on packages/preview/light-cv/0.1.0/template/cv-en.typ:
```diff
@@ -0,0 +1,167 @@
+      link: "https://linkedin.com/"
+    )
+  ),
+  profile-picture: "../media/avatar.jpeg"
```
Done

laurmaedje commented on packages/preview/light-cv/0.1.0/typst.toml:
```diff
@@ -0,0 +1,16 @@
+repository = "https://github.com/AnsgarLichter/cv-typst-template"
+keywords = ["cv", "curriculum vitae", "computer science", "application", "resume"]
+categories = ["cv"]
+exclude = ["example.png"]
```
There was a slight misunderstanding I think. The thumbnail `example.png` is automatically excluded. The one I was referring to is `template/media.image.png`, which will be included in the template bundle and cloned into the user's project, even though it is unused. Since it's only cutout of the proper `example.png`, it can even just be deleted rather than excluded, I think.

Other than that, this looks good to merge.

AnsgarLichter replied:
It's fixed now.  Thank you!

laurmaedje commented:
Thanks!
